airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/components.py

```diff
@@ -72,3 +73,58 @@ def extract_records(self, response: requests.Response) -> List[Mapping[str, Any]
         except requests.exceptions.JSONDecodeError:
             records = []
         return records
+
+
+class TicketAuditsRetriever(SimpleRetriever):
+    def __post_init__(self, parameters: Mapping[str, Any]):
+        super().__post_init__(parameters)
+
+        self.record_selector.transformations = []
+        self._start_date = self.config["start_date"]
+        self.response_list_name = self._parameters["data_path"]
+        self._cursor_field = self._parameters["cursor_field"]
+        self._start_date = self.config["start_date"]
+
+        self.cursor = DatetimeBasedCursor(
+            config=self.config,
+            parameters=self._parameters,
+            cursor_datetime_formats=[
+                "%Y-%m-%dT%H:%M:%SZ",
+            ],
+            datetime_format="%Y-%m-%dT%H:%M:%SZ",
+            cursor_field=self._cursor_field,
+            start_datetime=self._start_date,
+        )
+        self.stream_slicer = self.cursor
+
+    def _validate_response(self, response: requests.Response, stream_state: Mapping[str, Any]) -> bool:
+        """
+        Ticket Audits endpoint doesn't allow filtering by date, but all data sorted by descending.
+        This method used to stop making requests once we receive a response with cursor value greater than actual cursor.
+        This action decreases sync time as we don't filter extra records in parse response.
+        """
```

Comment:
Could we use `is_data_feed` option in `DatetimeBasedCursor` ?

Reply:
It was already added to low-code stream, but syncs still stuck

Reply:
removed custom component, added ` or not last_record` to `stop_condition`.